core/include/kernel/dt.h

```diff
@@ -35,7 +35,7 @@ struct dt_driver {
 	const void *driver;
 };
 
-#define __dt_driver __section(".rodata.dtdrv")
+#define __dt_driver __section(".rodata.dtdrv" __SECTION_FLAGS_RODATA)
```

Comment:
maybe remove since you defined it in the **compiler.h**.

Reply:
Ooops! It should be defined only once indeed.
Actually, I was not sure where the best place would be. I'd rather keep it in `dt.h`, no need to pollute `compiler.h`. I'll remove it from compiler.h if you don't mind.